PhysObjectExtractor/src/JetAnalyzer.cc

```diff
@@ -78,20 +105,47 @@ class JetAnalyzer : public edm::EDAnalyzer {
 JetAnalyzer::JetAnalyzer(const edm::ParameterSet& iConfig)
 {
 //now do what ever initialization is needed
-	jetInput = iConfig.getParameter<edm::InputTag>("InputCollection");
-	edm::Service<TFileService> fs;
-	mtree = fs->make<TTree>("Events", "Events");
-	
+  jetInput = iConfig.getParameter<edm::InputTag>("InputCollection");
+  edm::Service<TFileService> fs;
+  mtree = fs->make<TTree>("Events", "Events");
+  
+  isData = iConfig.getParameter<bool>("isData");
+  jecL1_ = iConfig.getParameter<edm::FileInPath>("jecL1Name").fullPath(); // JEC level payloads                     
+  jecL2_ = iConfig.getParameter<edm::FileInPath>("jecL2Name").fullPath(); // JEC level payloads                     
+  jecL3_ = iConfig.getParameter<edm::FileInPath>("jecL3Name").fullPath(); // JEC level payloads                     
+  jecRes_= iConfig.getParameter<edm::FileInPath>("jecResName").fullPath();
+  jecUncName_ = iConfig.getParameter<edm::FileInPath>("jecUncName").fullPath();      // JEC uncertainties                               
+
+  //Get the factorized jet corrector parameters.
+  jecPayloadNames_.push_back(jecL1_);
+  jecPayloadNames_.push_back(jecL2_);
+  jecPayloadNames_.push_back(jecL3_);
+  if( isData == true ) jecPayloadNames_.push_back(jecRes_);
+  std::vector<JetCorrectorParameters> vPar;
+  for ( std::vector<std::string>::const_iterator payloadBegin = jecPayloadNames_.begin(),
+	  payloadEnd = jecPayloadNames_.end(), ipayload = payloadBegin; ipayload != payloadEnd; ++ipayload ) {
+    JetCorrectorParameters pars(*ipayload);
+    vPar.push_back(pars);
+  }
+
+  // Make the FactorizedJetCorrector and Uncertainty                                                                                              
+  jec_ = boost::shared_ptr<FactorizedJetCorrector> ( new FactorizedJetCorrector(vPar) );
+  jecUnc_ = boost::shared_ptr<JetCorrectionUncertainty>( new JetCorrectionUncertainty(jecUncName_) );
 	
-	mtree->Branch("numberjet",&numjet);
```

Comment:
Need branch titles (for documentation) for all variables as was done for electrons